### PR TITLE
Added Picrew.me support

### PR DESCRIPTION
**Pull request by upsetdog**
Support for Picrew.me, with a French description as well.

### CODE REVIEW

**Comment by upsetdog** [comment]
Linked to issue #4301

File: websites/P/Picrew/presence.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        } else {
+          details = "creating a masterpiece";
+          state = picrewTitle;
+        }
+      } else 
+        details = "on Picrew";
+      
+  }
+
+  data.largeImageKey = "logo";
```
Comment (EncryptedDev) [inline websites/P/Picrew/presence.ts:44]: Move this to the data object on L9
```suggestion
```

File: websites/P/Picrew/presence.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+const presence: Presence = new Presence({
+  clientId: "886406987137576981",
+  injectOnComplete: true
+});
+
+presence.on("UpdateData", () => {
+  const {pathname} = window.location,
+
+   data: PresenceData = {};
```
Comment (EncryptedDev) [inline websites/P/Picrew/presence.ts:9]: Fix indentation here, but this is better

```suggestion
   data: PresenceData = {
     data.largeImageKey = "logo";
};
```

File: websites/P/Picrew/presence.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        }
+      } else 
+        details = "on Picrew";
+      
+  }
+
+  data.largeImageKey = "logo";
+  data.state = state;
+  data.details = details;
+  data.startTimestamp = Math.round(startTimestamp / 1000);
```
Comment (EncryptedDev) [inline websites/P/Picrew/presence.ts:47]: Move this to the data object on L9 as well
```suggestion
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/P/Picrew/presence.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      break;
+    case "/search":
+      details = "searching";
+      break;
+    case "/discovery":
+      details = "browsing the discovery page";
+      break;
+    default:
+      if (pathname.includes("/image_maker/")) {
+        const picrewId = pathname.match(/\/image_maker\/([0-9]{0,8})/)[1],
```
Comment (Slowlife01) [inline websites/P/Picrew/presence.ts:27]:
```suggestion
        const [, picrewId] = pathname.match(/\/image_maker\/([0-9]{0,8})/),
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]
add support for [support.picrew.me](https://support.picrew.me) too

File: websites/P/Picrew/presence.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+const presence: Presence = new Presence({
+  clientId: "886406987137576981"
+}),
+
+ { pathname } = window.location,
+  data: PresenceData = {
+    largeImageKey: "logo",
+    startTimestamp: Math.round(new Date().getTime() / 1000)
```
Comment (Slowlife01) [inline websites/P/Picrew/presence.ts:8]: No, I meant only startTimestamp

**Review by Bas950 — CHANGES_REQUESTED** [review]
Upload proof of it working and add more to the presence if possible.

**Comment by upsetdog** [comment]
> 
> 
> Upload proof of it working and add more to the presence if possible.

it works:
image
image
image
image

there's not really much more to add for functionalliy, it has enough at the moment. maybe in the future.
--

and i don't see a reason to add support.picrew.me, because users don't do anything on that sub-domain, but i added it anyway.

File: websites/P/Picrew/presence.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      largeImageKey: "logo",
+      startTimestamp
+    };
+
+  switch (pathname) {
+    case "/":
+      details = "on the homepage";
+      break;
+    case "/search":
+      details = "searching";
```
[the remark this answers is not in the document]
Comment (upsetdog) [reply websites/P/Picrew/presence.ts:27]: added
image

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Fix codefactor issue

**Pull request closed without merging**